plugin/pulpcore/plugin/changeset/pending_version.py

```diff
@@ -0,0 +1,50 @@
+from pulpcore.plugin.changeset.main import ChangeSet
+from pulpcore.plugin.tasking import WorkingDirectory
+from pulpcore.plugin.models import RepositoryVersion
+
+
+class PendingVersion:
+
+    def __init__(self, pending_content, repository, remote, sync_mode='mirror'):
+        self.pending_content = pending_content
+        self.repository = repository
+        self.remote = remote
+        self.sync_mode = sync_mode
+
+    def apply(self):
```

Comment:
I think `create()` is more appropriate than `apply()`.  I also think it would be appropriate for this method to return the newly created `RepositoryVersion` to the plugin.